File: src/main/java/org/springframework/samples/petclinic/service/interfaces/VisitService.java
package org.springframework.samples.petclinic.service.interfaces;

import org.springframework.samples.petclinic.dtos.ResponseData;
import org.springframework.samples.petclinic.dtos.VisitDTO;

import java.util.Collection;

public interface VisitService {
    ResponseData<String> saveVisit(VisitDTO visitDTO);

    ResponseData<String> deleteVisit(int visitId);

    ResponseData<Collection<VisitDTO>> getVisitsByPetId(int petId);

    ResponseData<Collection<VisitDTO>> getVisitsByDate(String dateString);

    ResponseData<Collection<VisitDTO>> getVisitsByDateRange(String startDate, String endDate);

    ResponseData<Collection<VisitDTO>> getVisitsByVetId(int vetId);
}

File: client/src/service/visit/visitService.js
import clinicRequest from '../clinicRequest';

const endpoint = '/visit';

const getVisitsByPetId = petId => {
  return clinicRequest.get(endpoint + `/pet-id/${petId}`);
};

const saveVisit = visit => {
  return clinicRequest.post(endpoint, visit);
};

const getVisitsByVetId = vetId => {
  return clinicRequest.get(endpoint + `/vet-id/${vetId}`);
};

const getVisitsByDate = dateString => {
  return clinicRequest.get(endpoint + `/visit-date/${dateString}`);
};

const getVisitsByDateRange = (startDate, endDate) => {
  const dateRange = { startDate, endDate };
  return clinicRequest.get(endpoint + '/visit-date', dateRange);
};

const visitService = {
  getVisitsByPetId,
  saveVisit,
  getVisitsByVetId,
  getVisitsByDate,
  getVisitsByDateRange
};
export default visitService;
